sdk/servicebus/service-bus/test/perf/service-bus-v7/send.ts

```diff
@@ -0,0 +1,139 @@
+/*
+# Overview
+Measures the maximum throughput of `sender.send()` in package `@azure/service-bus`.
+
+# Instructions
+1. Create a Service Bus namespace with `Tier=Premium` and `Messaging Units=4`.  It is recommended to use the largest possible namespace to allow maximum client throughput.
+2. Create a queue inside the namespace.
+3. Set env vars `SERVICEBUS_CONNECTION_STRING` and `SERVICEBUS_QUEUE_NAME`.
+4. `ts-node app.ts [maxInflightMessages] [totalMessages] [batchAPI]`
+5. Example: `ts-node app.ts 1000 1000000 true`
+ */
+
+import { ServiceBusClient, ServiceBusSender } from "@azure/service-bus";
+import delay from "delay";
+import moment from "moment";
+// Load the .env file if it exists
+import * as dotenv from "dotenv";
+dotenv.config();
+
+const _payload = Buffer.alloc(1024);
+const _start = moment();
+
+let _messages = 0;
+
+async function main(): Promise<void> {
+  // Endpoint=sb://<your-namespace>.servicebus.windows.net/;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=<shared-access-key>
+  const connectionString = process.env.SERVICEBUS_CONNECTION_STRING as string;
+  const entityPath = process.env.SERVICEBUS_QUEUE_NAME as string;
+
+  const maxInflight = process.argv.length > 2 ? parseInt(process.argv[2]) : 1;
+  const messages = process.argv.length > 3 ? parseInt(process.argv[3]) : 10;
+  const batchAPI = process.argv.length > 4 ? Boolean(process.argv[4]) : false;
+  log(`Maximum inflight messages: ${maxInflight}`);
+  log(`Total messages: ${messages}`);
+
+  const writeResultsPromise = WriteResults(messages);
+
+  await RunTest(connectionString, entityPath, maxInflight, messages, batchAPI);
+
+  await writeResultsPromise;
+}
+
+async function RunTest(
+  connectionString: string,
+  entityPath: string,
+  maxInflight: number,
+  messages: number,
+  batchAPI: boolean
+): Promise<void> {
+  const ns = new ServiceBusClient(connectionString);
+
+  const sender = ns.createSender(entityPath);
+
+  const promises: Promise<void>[] = [];
+
+  for (let i = 0; i < maxInflight; i++) {
+    const promise = ExecuteSendsAsync(sender, messages, batchAPI);
+    promises[i] = promise;
+  }
+
+  await Promise.all(promises);
+
+  await ns.close();
+}
+
+async function ExecuteSendsAsync(
+  sender: ServiceBusSender,
+  messages: number,
+  batchAPI: boolean
+): Promise<void> {
+  while (++_messages <= messages) {
```

Comment:
So I feel like I'd need to read this particular bit of code a few times to get what's happening. 

Is there any way this could be made a bit more clear? There's literally a variable called `messages` but it's not the messages it's...the current index of messages? 

Then we're incrementing messages in a while loop condition _twice_.

Reply:
My intention was to keep the program same as the track 1 one.
The outer while loop is same for both track 1 and track 2.

The 3-4 folds observation from the morning was from the single message API (else block) and not the batch API.

"messages" is the number of messages. Sure, I can update in both track 1 and track 2.

For the batch API, instead of sending a new message each time, I'm looping over to add the messages to the batch.
I can rework on the inner loop to make it clear.